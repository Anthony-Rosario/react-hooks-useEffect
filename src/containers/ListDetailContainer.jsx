/* eslint-disable react/prop-types */
import React from 'react';
import CharacterDetail from '../components/app/display/CharacterDetail';
import { useCharacter } from '../hooks/useCharacter';

const ListDetailContainer = () => {
  const [loading, character] = useCharacter();

  return (
    <div>
      {loading ? <h1>Loading...</h1>
        :
        <CharacterDetail character={character} />
      }
    </div>
  );
};

export default ListDetailContainer;
