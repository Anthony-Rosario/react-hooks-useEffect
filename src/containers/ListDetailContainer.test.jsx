/* eslint-disable max-len */
import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import ListDetailContainer from './ListDetailContainer';
import mockData from '../../fixtures/mockData';
import { MemoryRouter, Route } from 'react-router-dom';

const server = setupServer(
  rest.get('https://hey-arnold-api.herokuapp.com/api/v1/characters', (req, res, ctx) => {
    return res (ctx.json(mockData));
  })
);

describe('Hey Arnold detail container', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());
  it('should display the character that was clicked on based on its id', async () => {
    render(
      <MemoryRouter initialEntries={['/5da237699734fdcb7bef8f51']}><Route path="/:id"><ListDetailContainer/></Route> </MemoryRouter>);

    const loading = screen.getByText('Loading...');
    expect(loading).toMatchSnapshot();

    const figure = await screen.findByText('Arnold Shortman');

    expect(figure).toMatchSnapshot();
  });
});
