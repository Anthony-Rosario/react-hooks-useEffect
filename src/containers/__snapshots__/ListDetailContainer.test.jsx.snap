// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hey Arnold detail container should display the character that was clicked on based on its id 1`] = `
<h1>
  Loading...
</h1>
`;

exports[`Hey Arnold detail container should display the character that was clicked on based on its id 2`] = `
<figcaption>
  Arnold Shortman
</figcaption>
`;
